Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs

```diff
@@ -480,7 +534,9 @@ public ActionResult Problem(int id, bool official)
                 return this.RedirectToAction("Register", new { id = problem.ContestId, official });
             }
 
-            if (official && !this.ValidateContestIp(this.Request.UserHostAddress, problem.ContestId))
+            if (official &&
+                !problem.Contest.IsOnline &&
```

Comment:
Why do we exclude Online contests from Ip validation, the fact that they'll probably be opened, does not mean that no IP validation should happen.